woocommerce/woocommerce-gutenberg-products-block
# Update @woocmmerce/eslint-plugin to 2.0.0

This fixes the `npm run lint:js:report` error in CI. This was due to a conflict of eslint versions, where `@woocommerce/eslint-plugin` was using v7, and we (or another package) are using v8.

```
Oops! Something went wrong! :(

ESLint: 8.12.0

ESLint couldn't find the plugin "eslint-plugin-testing-library".
```

As part of this upgrade, `npm run lint:js` started to throw lots of errors. I've fixed most of these as part of this PR too

### Manual Testing

How to test the changes in this Pull Request:

1. `npm ci --no-optional && npm run lint:js:report` should be successful 
2. `npm run lint:js` should show 0 errors and 5 warnings
3. Best to check all other linting scripts the depends on eslint run successfully as well
